### PR TITLE
(IRO-1793) Added keyboard navigation to search autocomplete.

### PR DESCRIPTION
**Pull request by azhurauski-sc**
Added up, down, Enter and Escape keys handling.

### CODE REVIEW

File: components/SearchAutocomplete.tsx [hunk 1]
```diff
@@ -62,6 +63,7 @@ const SearchAutocomplete: FC<SearchAutocompleteProps> = ({
   const inputRef = useRef<HTMLInputElement>()
   const popoverRef = useRef<HTMLDivElement>()
   const { onOpen, onClose, isOpen } = useDisclosure()
+  const [focusedOption, setFocusedOption] = useState<SearchOption | null>(null)
```
Comment (brekk) [inline components/SearchAutocomplete.tsx:66]: Minor (and I'm working on an `eslint` plugin for this), but can we use the `$` prefixes on variables coming from `useState` / hooks generally?

File: components/SearchAutocomplete.tsx [hunk 2]
```diff
@@ -96,6 +103,47 @@ const SearchAutocomplete: FC<SearchAutocompleteProps> = ({
...
+      if ('options' in option) {
+        acc.push(...option.options)
+      } else {
+        acc.push(option)
+      }
+      return acc
+    }, [])
+    const focusedIndex = focusedOption
+      ? focusableOptions.indexOf(focusedOption)
+      : -1
```
Comment (brekk) [inline components/SearchAutocomplete.tsx:124]: `indexOf` returns -1 by default, so I think you can always just do `focusableOptions.indexOf(focusedOption)` (e.g. `"cool".indexOf(null)` )

File: components/SearchAutocomplete.tsx [hunk 3]
```diff
@@ -141,6 +190,7 @@ const SearchAutocomplete: FC<SearchAutocompleteProps> = ({
               if ('options' in option) {
                 return option.options.length ? (
                   <Group
+                    key={`options-${option.label}-group`}
```
Comment (brekk) [inline components/SearchAutocomplete.tsx:193]: Not a blocker, but is this needed? I'd think just `option.label` would work?

**Review by brekk — APPROVED** [review]
Some minor suggestions, no blockers

**Review by ndemidovich-sc — APPROVED** [review]

File: components/SearchAutocomplete.tsx (hunk 1 above)
Comment (azhurauski-sc) [reply components/SearchAutocomplete.tsx:66]: Sure, didn't used to use it, but will try to remember. Having eslint rule would be great.

**Pull request merged**